2024/OSFC/enabling_coreboot_on_talosII.md

```diff
@@ -0,0 +1,1046 @@
+
+There are only 2 .initfile files, probably someone forgot to remove them from
+the tree.
+
+---
+
+# Why?
+
+Example of initfile in converted form:
+
+.center[.image-75[![](/img/initfile1.png)]]
+
+.footnote[Source: https://git.raptorcs.com/git/talos-hostboot/tree/src/import/chips/p9/procedures/hwp/initfiles/p9_npu_scom.C?id=a2ddbf3150e2c02ccc904b25d6650c9932a8a841]
+
+???
+
+There are four different ways of writing 0 and three ways of writing 1 on this
+slide.
+
+---
+
+# Why?
+
+Flow of every initfile is similar: read some unnamed register, check CPU
+version and attributes, write magic number back, repeat for the next register.
+
+.center[.image-100[![](/img/initfile2.png)]]
+
+.footnote[Source: https://git.raptorcs.com/git/talos-hostboot/tree/src/import/chips/p9/procedures/hwp/initfiles/p9_npu_scom.C?id=a2ddbf3150e2c02ccc904b25d6650c9932a8a841]
+
+???
+
+Most of the registers are documented, but not all.
+
+--
+
+Can you spot the difference between statements of middle level `if`s?
+
+???
+
+There is no difference!
+
+---
+
+# Hardware
+
+Talos II:
+
+.center[.image-80[![](/img/talos2.png)]]
+
+.footnote[Source: https://raptorcs.com/content/TL2MB1/intro.html]
+
+???
+
+SATA controller - not open, version without it is also available.
+
+2 CPUs, lots of RAM slots (registered DDR4 with ECC), lots of PCIe
+
+---
+
+# Hardware
+
+Talos II Lite:
+
+.center[.image-80[![](/img/talos2_lite.png)]]
+
+.footnote[Source: https://raptorcs.com/content/TL1MB1/intro.html]
+
+???
+
+Less slots, SATA not an option
+
+---
+
+# Hardware
+
+Specifications:
+
+- 2 POWER9-compatible CPU sockets
+  - Talos II Lite: 1 socket
+- EATX form factor
+- 16 DDR4 ECC registered RAM slots
+  - Talos II Lite: 8 slots
+- 3 PCIe 4.0 x16 slots
+  - Talos II Lite: 1 slots
+- 2 PCIe 4.0 x8 slots
+  - Talos II Lite: 1 slots
+- 2 Broadcom Gigabit Ethernet ports, one shared with BMC
+- 1 Microsemi SAS 3.0 controller (**optional**)
+  - Talos II Lite: n/a
+- 4 USB 3.0 ports, 1 USB 2.0 port
+- 1 ASpeed BMC with OpenBMC
+- 1 VGA video port
+
+---
+
+# Hardware
+
+POWER9 chip types:
+
+- Scale Out (Nimbus) with directly attached memory, used on Talos II
+- Scale Up (Cumulus) with memory connected through memory controllers called
+  Centaurs
+
+Nimbus modules (packages):
+
+- Sforza - 50 mm × 50 mm, 4 DDR4, 48 PCIe lanes, 1 XBus 4B, used on
+  Talos&nbsp;II
+- Monza - 68.5 mm × 68.5 mm, 8 DDR4, 34 PCIe lanes, 1 XBus 4B, 48 OpenCAPI lanes
+- LaGrange - 68.5 mm × 68.5 mm, 8 DDR4, 42 PCIe lanes, 2 XBus 4B, 16 OpenCAPI
+  lanes
+
+.footnote[Source: https://en.wikipedia.org/wiki/POWER9#Chip_types]
+
+???
+
+Talos II uses least server-like option.
+
+XBus connects two CPUs. There is only one XBus for Sforza, so max 2 CPUs on
+board.
+
+---
+
+# Hardware
+
+Many cores of SoC:
+
+.center[.image-80[![](/img/power9_ppe_instances.png)]]
+
+.footnote[Source: https://wiki.raptorcs.com/w/images/c/ce/POWER9_um_OpenPOWER_v21_10OCT2019_pub.pdf]
+
+???
+
+PPE - Programmable PowerPC-lite Engines
```

Comment:
I won't have enough time to mention everything I want :slightly_smiling_face: But since I wanted to mention that those are PPC I added exact versions to my notes, in case anyone asks.